projects/ui/src/Utility/utility.ts

```diff
@@ -167,3 +167,19 @@ export const useInArea = (paths: string[]) => {
     });
   }, [routerLocation.pathname, paths]);
 };
+
+export const shallowEquals = (
+  a: Record<string, any> | undefined,
+  b: Record<string, any> | undefined
+) => {
+  if (!a && !b) {
+    return true;
+  }
+  if (!a || !b) {
+    return false;
+  }
+  return (
+    Object.entries(a).every(([k, v]) => b[k] === v) &&
```

Comment:
This feels needlessly duplicated.  It's slight, but terser would just be to compare the lengths above, like. for the `(!a || !b)` to do:  `(a?.length !== b?.length)`, and then when we get to this check then they must be the same length and we're only doing 1 iteration